src/ctap2/commands/client_pin.rs

```diff
@@ -148,18 +181,32 @@ impl<'de> Deserialize<'de> for ClientPinResponse {
                             pin_token = map.next_value()?;
                         }
                         3 => {
-                            if retries.is_some() {
-                                return Err(SerdeError::duplicate_field("retries"));
+                            if pin_retries.is_some() {
+                                return Err(SerdeError::duplicate_field("pin_retries"));
+                            }
+                            pin_retries = Some(map.next_value()?);
+                        }
+                        4 => {
+                            if power_cycle_state.is_some() {
+                                return Err(SerdeError::duplicate_field("power_cycle_state"));
+                            }
+                            power_cycle_state = Some(map.next_value()?);
+                        }
+                        5 => {
+                            if uv_retries.is_some() {
+                                return Err(SerdeError::duplicate_field("uv_retries"));
                             }
-                            retries = Some(map.next_value()?);
+                            uv_retries = Some(map.next_value()?);
                         }
                         k => return Err(M::Error::custom(format!("unexpected key: {:?}", k))),
```

Comment:
Is an unexpected key necessarily an error? What if CTAP2.2 extends the response with new fields?

Reply:
Yes, we should probably do here the same as we do in GetInfo, and only log it and continue.